core/templates/dev/head/pages/exploration-player-page/suggestion-modal-for-learner-local-view/suggestion-modal-for-exploration-player.service.ts

```diff
@@ -61,10 +63,29 @@ angular.module('oppia').factory('SuggestionModalForExplorationPlayerService', [
             $timeout(function() {
               $scope.showEditor = true;
             }, 500);
-
             $scope.cancelSuggestion = function() {
               SuggestionModalService.cancelSuggestion($uibModalInstance);
             };
+            $scope.$on('changed', function(evt, data) {
+              var compare2 = data;
+              var compare = $scope.originalHtml;
+              var undoRegExp = new RegExp(['<div class="oppia-rte-component',
+                '-container" type="oppia-noninteractive-.+?\\b[^>]*>']
+                .join(''), 'gm');
+              compare2 = compare2.replace(undoRegExp, '');
+              compare2 = compare2.replace(/<\/div>/gm, '');
+              compare2 = compare2.replace(/^\s*[\r\n]/gm, '').trim();
+              compare = compare.replace(/^\s*[\r\n]/gm, '').trim();
+              if (compare === compare2) {
+                $scope.$apply(function() {
```

Comment:
Why do you need the $scope.$apply()?

Reply:
The ng-disabled doesn't seem to pick up the last changed value of `$scope.ifContentChanged` so I tried doing this.